packages/reports/tests/integration/components/column-selector-test.ts

```diff
@@ -0,0 +1,135 @@
+    const groupColumns = findAll('.column-selector__column').map((e) => e?.textContent?.trim());
+    assert.deepEqual(
+      groupColumns,
+      ['EventId', 'Parent Event Id', 'Product Family', 'Property'],
+      '`column-selector` expands a column group correctly'
+    );
+  });
+
+  test('search', async function (this: TestContext, assert) {
+    await render(TEMPLATE);
+
+    assert.dom('.column-selector__column').doesNotExist('All column groups are collapsed');
+    await fillIn('.column-selector__search-input', 'age');
+
+    const groups = findAll('.grouped-list__group-header').map((e) => e?.textContent?.trim());
+    assert.deepEqual(groups, ['test (1)'], '`column-selector` renders search column groups correctly');
+
+    const groupColumns = findAll('.column-selector__column').map((e) => e?.textContent?.trim());
+    assert.deepEqual(groupColumns, ['Age'], '`column-selector` expands search column group correctly');
+
+    await fillIn('.column-selector__search-input', 'not a column');
+    assert
+      .dom('.grouped-list__group-header')
+      .doesNotExist('`column-selector` does not show column groups when no search column is found');
+    assert
+      .dom('.column-selector__column')
+      .doesNotExist('`column-selector` does not show columns when no search column is found');
+    assert.dom('.column-selector__no-match').exists('`column-selector` shows a message when no search column is found');
+  });
+
+  test('onAddColumn', async function (this: TestContext, assert) {
+    assert.expect(1);
+
+    this.set('onAddColumn', (column: ColumnMetadataModel) => {
+      assert.equal(column.name, 'Age', '`column-selector` calls `onAddColumn` when column is clicked');
+    });
+
+    await render(TEMPLATE);
+    await fillIn('.column-selector__search-input', 'age');
+    await click('.column-selector__add-column-btn');
+  });
+
+  test('onAddFilter', async function (this: TestContext, assert) {
+    assert.expect(1);
+
+    this.set('onAddFilter', (column: ColumnMetadataModel) => {
+      assert.equal(column.name, 'Age', '`column-selector` calls `onAddFilter` when filter is clicked');
+    });
+
+    await render(TEMPLATE);
+    await fillIn('.column-selector__search-input', 'age');
+    await click('.column-selector__add-filter-btn');
+  });
+
+  test('tooltip', async function (this: TestContext, assert) {
+    await render(TEMPLATE);
+
+    assertTooltipNotRendered(assert);
+    this.server.get(`${config.navi.dataSources[0].uri}/v1/dimensions/age`, function () {
+      return {
+        category: 'test',
+        name: 'age',
+        longName: 'Age',
+        type: 'string',
+        description: 'my description',
+      };
+    });
+
+    await fillIn('.column-selector__search-input', 'age');
+    await triggerEvent('.column-selector__column-info', 'mouseenter');
+
+    assertTooltipRendered(assert);
+    assertTooltipContent(assert, {
+      contentString: 'my description',
+    });
```

Comment:
Test no description case?

Reply:
It is built in: https://github.com/sir-dunxalot/ember-tooltips/blob/3.4.7/addon-test-support/dom/assertions/assert-tooltip-content.js#L27